### DISCUSSION TITLE
Send random template from selected pool of templates in a campaign

### DISCUSSION THREAD

**Discussion by Luke-From-IT** [gophish/gophish#2732] [Ideas]
Hi,

I think a great idea would be to allocate a set of templates to use in a campaign, where a random one out of the selected pool is sent to the users within that campaign. I have attached a basic mock-up of how this could work in case my description doesn't make sense.

![image](https://user-images.githubusercontent.com/46346663/213185981-b81a095b-5ae0-45ae-8ff8-a4ca422f60b4.png)

**Comment by Luke-From-IT** [comment]
@glennzw, I'm not sure if you get notifications for discussions, so I hope you don't mind being tagged.

**Comment by glennzw** [comment]
Hi @Luke-From-IT, ah you're right I don't have notifications on for Discussions - I see there's a bunch of stuff in here I've missed!

Great idea on the randomised templates. Technically it might be quite tricky but I'll give it some thought. Also, two questions/thoughts:

- Would you randomise the Landing Page too? I'm not sure how this would work unless you have a pool of matching from each (i.e. it'd be weird to get a random Facebook email that links to a DropBox login).
- From a usefulness point of view I'm not sure if sending different templates in the same campaign is will accurately represent the target group. e.g if one person gets an obvious Nigerian Prince scam and reports it, but another gets a very well crafted 'Your inbox is full' email and falls for it.

What I favour is to send small representative campaigns - e.g 10% of staff from each department. For your use case you could send multiple small campaigns, and if need be merge the results (via API or .csv). You could even Pythonify your randomiser using the API.

**Reply by Luke-From-IT** [reply]
Hmm, that's a great question; in my use case, we don't use a bunch of landing pages. We have one "You've been phished" landing page with a link to our training system (currently using the API to enrol them on our courses if they've been caught).

Although I see the use case for multiple as well dependent on the template used, one thought I had was to have the option to set a "default landing page" for each template within the template section, which the campaign itself could override.

I did think small campaigns could be useful, but they were also quite chaotic and admin-heavy.

The thing we are trying to avoid is teams alerting other people about the phishing email coming from KnowBe4 (we are currently trying to replace KB4 with GoPhish). We have campaigns that send randomised phishing templates out in order to avoid people being alerted.

**Comment by glennzw** [comment]
I'm all for replacing KB4 :)

Understood for your use case - it'd make sense if you use the same Landing Page for all email sims. I think leveraging the API is probably the best bet here. My initial thought would be:

- Create a command line utility allowing you to choose which _N_ Email Templates you want to randomly use.
- Randomly divide _E_ employees by _N_ templates, and create a new campaign for each
- Dispatch emails
- Update the gophish database 'merging' the multiple campaigns into one

I'll have to have a look under the hood to see if that's actually technically possible.

**Comment by glennzw** [comment]
I run a 'pro' gophish service which might interest you too - https://kapenta.io

**Comment by RaveNN-0** [comment]
Any updates on this?
